### PR TITLE
feat: Add secret resource to enforce policies

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

File: src/resources/secret.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        (error, response: ResourceDeclareResponse) => {
+          if (error) {
+            reject(fromGrpcError(error));
+          } else {
+            resolve(resource);
+          }
+        }
+      )
+    })
+  }
```
Comment (tjholm) [inline src/resources/secret.ts:54]: missing newline

File: src/resources/secret.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            return [...actions, Action.SECRETACCESS];
+          default:
+            throw new Error(
+              `unknown secret permission ${perm}, supported permissions are ${everything.join(
+                ', '
+              )}`
+            );
+        }
+      }, []);
+  }
```
Comment (tjholm) [inline src/resources/secret.ts:70]: missing newline

**Review by tjholm — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
   ],
   "dependencies": {
     "@grpc/grpc-js": "^1.1.7",
-    "@nitric/api": "^0.13.0",
+    "@nitric/api": "rc-latest",
```
Comment (davemooreuws) [inline package.json:30]: Not sure if this should be set to the latest rc version.

**Review by davemooreuws — CHANGES_REQUESTED** [review]
Looks good. Just a QA

File: src/resources/secret.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          }
+        }
+      )
+    })
+  }
+
+  protected permsToActions(...perms: SecretPermission[]): ActionsList {
+      return perms.reduce((actions, perm) => {
+        switch(perm) {
+          case 'put':
```
Comment (asalkeld) [inline src/resources/secret.ts:59]: or "update"? for("put") sounds a bit odd

Comment (tjholm) [reply src/resources/secret.ts:59]: @asalkeld Using 'put' at the moment as the secret stores we're integration with are all append, so we're actually just adding a new secret value which will be rotated in at the `latest` secret value. I think update is fine as well, but in some ways implies that the old secret value will no longer exist, but it does still hang around.

File: package.json (hunk 3 above)
Comment (HomelessDinosaur) [reply package.json]: Made change to lock version to 0.14.0-rc.4

**Review by davemooreuws — APPROVED** [review]

**Pull request merged**